go/pipeline/base_batch.go

```diff
@@ -0,0 +1,6454 @@
+//
+// See [valkey.io] for details.
+//
+// Since:
+//
+//	Valkey 6.2.0 and above.
+//
+// Parameters:
+//
+//	destinationKey - The key of the sorted set to store the result.
+//	sourceKey - The key of the sorted set to search.
+//	searchFrom - The query's center point options, could be one of:
+//		 - `MemberOrigin` to use the position of the given existing member in the sorted
+//	          set.
+//		 - `CoordOrigin` to use the given longitude and latitude coordinates.
+//	searchByShape - The query's shape options:
+//		 - `BYRADIUS` to search inside circular area according to given radius.
+//		 - `BYBOX` to search inside an axis-aligned rectangle, determined by height and width.
+//	infoOptions - The optional inputs to request additional information.
+//
+// Command Response:
+//
+//	The number of elements in the resulting set.
+//
+// [valkey.io]: https://valkey.io/commands/geosearchstore/
+func (b *BaseBatch[T]) GeoSearchStoreWithInfoOptions(
+	destinationKey string,
+	sourceKey string,
+	searchFrom options.GeoSearchOrigin,
+	searchByShape options.GeoSearchShape,
+	infoOptions options.GeoSearchStoreInfoOptions,
+) *T {
+	return b.GeoSearchStoreWithFullOptions(
+		destinationKey,
+		sourceKey,
+		searchFrom,
+		searchByShape,
+		*options.NewGeoSearchResultOptions(),
+		infoOptions,
+	)
+}
+
+// Loads a library to Valkey.
+//
+// Since:
+//
+//	Valkey 7.0 and above.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	libraryCode - The source code that implements the library.
+//	replace - Whether the given library should overwrite a library with the same name if it already exists.
+//
+// Command Response:
+//
+//	The library name that was loaded.
+//
+// [valkey.io]: https://valkey.io/commands/function-load/
+func (b *BaseBatch[T]) FunctionLoad(libraryCode string, replace bool) *T {
+	args := []string{}
+	if replace {
+		args = append(args, constants.ReplaceKeyword)
+	}
+	args = append(args, libraryCode)
+	return b.addCmdAndTypeChecker(C.FunctionLoad, args, reflect.String, false)
+}
+
+// Deletes a library and all its functions.
+//
+// Since:
+//
+//	Valkey 7.0 and above.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	libName - The library name to delete.
+//
+// Command Response:
+//
+//	"OK" if the library exists, otherwise an error is thrown.
+//
+// [valkey.io]: https://valkey.io/commands/function-delete/
+func (b *BaseBatch[T]) FunctionDelete(libName string) *T {
+	return b.addCmdAndTypeChecker(C.FunctionDelete, []string{libName}, reflect.String, false)
+}
+
+// Deletes all the libraries.
+//
+// Since:
+//
+//	Valkey 7.0 and above.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	async - Whether the libraries should be deleted asynchronously.
+//
+// Command Response:
+//
+//	OK if the libraries were deleted successfully.
+//
+// [valkey.io]: https://valkey.io/commands/function-flush/
+func (b *BaseBatch[T]) FunctionFlush(async bool) *T {
```

Comment:
This signature seems to differ from `FunctionFlush(ctx context.Context) (string, error)`
in `ScriptingAndFunctionBaseCommands` where this func takes an `async` parameter. All other commands share the same signature (minus the need for `ctx context.Context`)